test/role.test.js

```diff
@@ -208,4 +210,64 @@ describe('role model', function() {
 
   });
 
+  describe('listByPrincipalType', function() {
+    var sandbox;
+
+    beforeEach(function() {
+      sandbox = sinon.sandbox.create();
+    });
+
+    afterEach(function() {
+      sandbox.restore();
+    });
+
+    it('should fetch all models assigned to the role', function(done) {
+      var principalTypesToModels = {};
+      var runs = 0;
+      var mappings;
+
+      principalTypesToModels[RoleMapping.USER] = User;
+      principalTypesToModels[RoleMapping.APPLICATION] = Application;
+      principalTypesToModels[RoleMapping.ROLE] = Role;
+
+      mappings = Object.keys(principalTypesToModels);
+
+      mappings.forEach(function(principalType) {
+        var Model = principalTypesToModels[principalType];
+        Model.create({name:'test', email:'x@y.com', password: 'foobar'}, function(err, model) {
+          Role.create({name:'testRole'}, function(err, role) {
+            role.principals.create({principalType: principalType, principalId: model.id}, function(err, p) {
+              var pluralName = Model.pluralModelName.toLowerCase();
+              role[pluralName](function(err, models) {
+                assert(!err);
+                assert.equal(models.length, 1);
+                if (++runs === mappings.length) {
+                  done();
+                }
+              });
+            });
+          });
+        });
+      });
+    });
+
+    it('should apply query', function(done) {
+      User.create({name: 'Raymond', email: 'x@y.com', password: 'foobar'}, function(err, user) {
+        Role.create({name: 'userRole'}, function(err, role) {
+          role.principals.create({principalType: RoleMapping.USER, principalId: user.id}, function(err, p) {
+            var query = {fields:['id', 'name']};
+            sandbox.spy(User, 'find');
```

Comment:
A spy is used to examine the calls of a function and the arguments passed to it. In unit tests, its important to test with laser focus. Instead of relying on the result of the .find function, we only need to test that the correct input was passed to it. I would suggest an iterative approach to adding sinon so that some tests can benefit while others are being converted. I could set it up in a way that makes it easy to use in other tests, and implement in some of the other tests when I have time. I saw a few other places that could benefit from it.